drivers/hmis/app/models/hmis/hud/client.rb

```diff
@@ -91,4 +118,101 @@ def self.sort_by_option(option)
       raise NotImplementedError
     end
   end
+
+  def self.data_quality_enum_map_for(type = :other)
+    desc_map = {
+      name: {
+        full: ::HUD.name_data_quality(1),
+        partial: ::HUD.name_data_quality(2),
+      },
+      ssn: {
+        full: ::HUD.ssn_data_quality(1),
+        partial: ::HUD.ssn_data_quality(2),
+      },
+      dob: {
+        full: ::HUD.dob_data_quality(1),
+        partial: ::HUD.dob_data_quality(1),
+      },
+      other: {
+        full: 'Full value reported',
+        partial: 'Partial value reported',
+      },
+    }
+    desc_text = desc_map[type] || desc_map[:other]
+
+    Hmis::FieldMap.new(
+      [
+        {
+          key: :full,
+          value: 1,
+          desc: desc_text[:full],
+        },
+        {
+          key: :partial,
+          value: 2,
+          desc: desc_text[:partial],
+        },
+      ],
+    )
+  end
+
+  def self.name_data_quality_enum_map
+    data_quality_enum_map_for(:name)
+  end
+
+  def self.ssn_data_quality_enum_map
+    data_quality_enum_map_for(:ssn)
+  end
+
+  def self.dob_data_quality_enum_map
+    data_quality_enum_map_for(:dob)
+  end
+
+  def self.race_enum_map
+    Hmis::FieldMap.new(
+      ::HUD.races.except('RaceNone').map do |field, desc|
+        {
+          key: field,
+          value: field,
+          desc: desc,
+        }
+      end,
+    )
+  end
+
+  def self.gender_enum_map
+    Hmis::FieldMap.new(
+      ::HUD.genders.except(8, 9, 99).map do |value, desc|
+        {
+          key: ::HUD.gender_id_to_field_name[value],
+          value: value,
+          desc: desc,
+        }
+      end,
+    )
+  end
+
+  def self.ethnicity_enum_map
+    Hmis::FieldMap.new(
+      ::HUD.ethnicities.slice(0, 1).map do |value, desc|
+        {
+          key: desc.split('/').first,
+          value: value,
+          desc: desc,
+        }
+      end,
+    )
+  end
+
+  def self.veteran_status_enum_map
+    Hmis::FieldMap.new(
+      ::HUD.no_yes_reasons_for_missing_data_options.slice(0, 1).map do |value, desc|
+        {
+          key: desc,
+          value: value,
+          desc: desc,
+        }
+      end,
+    )
```

Comment:
This pattern is going to be repeated a bunch of times.  probably worth factoring it out into a concern so you can just add `no_yes_options` here, and in the future, or maybe `no_yes_field_map`.

```
def self. no_yes_field_map
  ::HUD.no_yes_reasons_for_missing_data_options.slice(0, 1).map do |value, desc|
         {
           key: desc,
           value: value,
           desc: desc,
         }
       end,
     )
end
```

Reply:
or maybe that's a factory method  on `FieldMap` so you can just call: `Hmis::FieldMap.no_yes`.